### ISSUE TITLE
add more unittest to test all the changes accompanying verkle

### ISSUE THREAD

**Issue by tanishqjasoria** [NethermindEth/nethermind#6698]
- [ ] New state tree
- [ ] Proof generation and verification
- [ ] Stateless gas changes
- [ ] Execution Witness generation
- [ ] Stateless executing blocks from witness generated by the nethermind client
- [ ] Verkle Sync

**Comment by ebukizy1** [comment]
please will love to work on this issue assign me and give me the necessary information  that will also help will get this done asap

**Comment by guha-rahul** [comment]
Maybe we can tackle this together @ebukizy1 .

**Comment by ebukizy1** [comment]
okay .. that would be nice

**Comment by tanishqjasoria** [comment]
yes, you can pick this up if you want. there are two main categories for test that we need.
1. Stateless Gas Changes - you can refer to these https://github.com/NethermindEth/nethermind/tree/feature/verkle/src/Nethermind/Nethermind.Evm.Test/Verkle and then the spec https://eips.ethereum.org/EIPS/eip-4762 and add more tests to cover the changes in gas cost.
2. Block Producing/Processinf tests - here you can find engine module tests [here](https://github.com/NethermindEth/nethermind/tree/feature/verkle/src/Nethermind/Nethermind.Merge.Plugin.Test) to understand how to write these tests.

**Comment by rishabhRsinghvi** [comment]
@tanishqjasoria If it's not done yet, I can do it?

**Comment by ScottyDavies** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/nethermind).

### My background and how it can be leveraged
I am  blockchain developer with years of experience


### How I plan on tackling this issue
I would approach this issue with following steps

Analysis:
I understand Ethereum state management, proofs, and stateless execution.
I wound analyze existing client architectures
Architecture Design:
Create a modular, scalable architecture for the new requirements.
Identify key components and their interfaces.
New State Tree:
Design and implement the new state tree (e.g., Verkle Tree).
Ensure security, efficiency, and scalability.
Proof System:
Develop components for generating and verifying cryptographic proofs.
Optimize for performance and scalability.
Stateless Gas Changes:
Analyze the gas model and implement stateless changes.
Integrate with the new state tree and proof system.
Execution Witness:
Develop components to generate execution witnesses.
Integrate with the client's execution engine and the new state tree.
Stateless Block Execution:
Integrate the Nethermind-generated witness.
Enable secure, efficient stateless block execution.
Verkle Sync:
Design and implement the Verkle Sync mechanism.
Leverage the new state tree and proof system.
Testing and Deployment:
Thoroughly test the components and the overall system.
Develop deployment scripts and configuration.
Plan a gradual rollout strategy.

**Comment by tanishqjasoria** [comment]
everything you described is already implemented in the client. this issues is only about writing unittests for the changes.

**Comment by RomThpt** [comment]
Is this issue still available?

**Comment by mymiracle0118** [comment]
I’d like to resolve this.

**Comment by mimisavage** [comment]
I'd like to handle this

**Comment by kamilchodola** [comment]
cc @dmitriy-b

**Comment by olisaagbafor** [comment]
Let me try this one!
